src/Plugin/RulesAction/RulesComponentAction.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\RulesAction\RulesComponentAction.
+ */
+
+namespace Drupal\rules\Plugin\RulesAction;
+
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\rules\Core\RulesActionBase;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a generic 'Execute Rules component' action.
+ *
+ * @RulesAction(
+ *   id = "rules_component",
+ *   deriver = "Drupal\rules\Plugin\RulesAction\RulesComponentActionDeriver"
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ */
+class RulesComponentAction extends RulesActionBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The storage of rules components.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $storage;
+
+  /**
+   * The ID of the rules component config entity.
+   *
+   * @var string
+   */
+  protected $componentId;
```

Comment:
no, you are right. I just seemed duplicating as it already part of the the plugin definition. I guess it's a short-cut? It's a bit duplicating data now, so a method would be nicer - but no biggie.